packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/NetCDFController.java

```diff
@@ -0,0 +1,232 @@
+		@ApiResponse(responseCode = "500", description = "There was an issue retrieving the netCDFs", content = @Content)
+	})
+	public ResponseEntity<List<NetCDF>> getNetCDFs(
+		@RequestParam(name = "page-size", defaultValue = "100", required = false) final Integer pageSize,
+		@RequestParam(name = "page", defaultValue = "0", required = false) final Integer page) {
+		try {
+			return ResponseEntity.ok(netCDFService.getNetCDFs(page, pageSize));
+		} catch (Exception e) {
+			final String error = "Unable to get netCDFs";
+			log.error(error, e);
+			throw new ResponseStatusException(
+				org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR,
+				error);
+		}
+	}
+
+	@PostMapping
+	@Secured(Roles.USER)
+	@Operation(summary = "Creates a new netCDF")
+	@ApiResponses(value = {
+		@ApiResponse(responseCode = "201", description = "NetCDF created.", content = @Content(mediaType = "application/json", schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = NetCDF.class))),
+		@ApiResponse(responseCode = "500", description = "There was an issue creating the netCDF", content = @Content)
+	})
+	public ResponseEntity<NetCDF> createNetCDF(@RequestBody NetCDF netCDF) {
+		try {
+			return ResponseEntity.status(HttpStatus.CREATED).body(netCDFService.createNetCDF(netCDF));
+		} catch (Exception e) {
+			final String error = "Unable to create netCDF";
+			log.error(error, e);
+			throw new ResponseStatusException(
+				org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR,
+				error);
+		}
+	}
+
+	@GetMapping("/{id}")
+	@Secured(Roles.USER)
+	@Operation(summary = "Gets an netCDF by ID")
+	@ApiResponses(value = {
+		@ApiResponse(responseCode = "200", description = "NetCDF retrieved.", content = @Content(mediaType = "application/json", schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = NetCDF.class))),
+		@ApiResponse(responseCode = "204", description = "NetCDF not found", content = @Content),
+		@ApiResponse(responseCode = "500", description = "There was an issue retrieving the netCDF", content = @Content)
+	})
+	public ResponseEntity<NetCDF> getNetCDF(@PathVariable("id") UUID netCDFId) {
+		try {
+			Optional<NetCDF> netCDF = netCDFService.getNetCDF(netCDFId);
+			if (netCDF.isEmpty()) {
+				return ResponseEntity.noContent().build();
+			}
+			return ResponseEntity.ok(netCDF.get());
+		} catch (Exception e) {
+			final String error = "Unable to get netCDF";
+			log.error(error, e);
+			throw new ResponseStatusException(
+				org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR,
+				error);
+		}
+	}
+
+	@PutMapping("/{id}")
+	@Secured(Roles.USER)
+	@Operation(summary = "Updates an netCDF")
+	@ApiResponses(value = {
+		@ApiResponse(responseCode = "200", description = "NetCDF updated.", content = @Content(mediaType = "application/json", schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = NetCDF.class))),
+		@ApiResponse(responseCode = "204", description = "NetCDF not found", content = @Content),
```

Comment:
This case is returning a `ResponseEntity.notFound().build()` which is `404`

Reply:
I copied DocumentController (or something similar) so that is the origin of that error - I was confused but assumed it was on purpose